### ISSUE TITLE
`use_build_context_synchronously` false positive

### ISSUE THREAD

**Issue by ahmednfwela** [dart-lang/linter#4331]
**Describe the issue**
lint `use_build_context_synchronously` is giving me a false positive

**To Reproduce**
this gives false positive
```dart
Future<void> test(BuildContext context) async {
  await Future.delayed(const Duration(seconds: 1));
  if (!context.mounted) return; //lint `use_build_context_synchronously`
}
```
but this works:
```dart
Future<void> test(BuildContext context) async {
  await Future.delayed(const Duration(seconds: 1));
  if (context.mounted) { //ok
        
  } else {
    return;
  }
}
```

**Expected behavior**
I should get no lints at all

**Additional context**
`flutter_lints: ^2.0.0`

**Comment by ahmednfwela** [comment]
may be fixable by https://github.com/dart-lang/linter/pull/4323
